(#21) Handle multiple rsan IPs and Undef

jarretlavallee opened the pull request:
Prior to this commit, when the RSAN IP was empty or had multiple entries
in the PuppetDB query, the catalog would fail to compile. This commit
allows for empty and multiple returned items from the PuppetDB query.

Resolves #21

Commit:
(#21) Handle multiple rsan IPs and Undef

Prior to this commit, when the RSAN IP was empty or had multiple entries
in the PuppetDB query, the catalog would fail to compile. This commit
allows for empty and multiple returned items from the PuppetDB query.

dylanratcliffe commented on manifests/exporter.pp:
```diff
@@ -7,7 +7,7 @@
 class rsan::exporter (
   $rsan_host = undef,
-  String $rsanip = rsan::get_rsan_ip(),
+  Array $rsanip = rsan::get_rsan_ip(),
```
Now that this is an array I'd recommend considering changing the naming e.g.

* Things that return an array should be plural
* `rsan::get_rsan_ip` actually technically gets the IPs of all "importers", not all RSAN nodes (i.e. not exporters) so possibly the naming could be improved here
* `$rsanip` in a way variable doesn't contain the RSAN IP, or at least that's not how it's used. What it really contains is a list of IPs that are allowed to mount the exported directories. Maybe we could make the name of this more clearly articulate that

jarretlavallee replied:
Great point. Updated

dylanratcliffe commented on manifests/exporter.pp:
```diff
@@ -19,21 +19,26 @@
     server_enabled => true
   }
 
+  $_rsan_clients = $rsanip.reduce('') |$memo, $ip| {
```
Can we get a comment here on what it does. Reduce functions are very hard to get your head around for many people

jarretlavallee replied:
Absolutely. I will add a comment in there. If there is a pattern that is more common or easier to read, let me know and I can change it.

dylanratcliffe reviewed (commented):
Overall I think this is a solid improvement. Added some comments around naming and things though which could improve it further

Commit:
(#21) Update comments and naming

This commit updates the names of the function that gets IP addresses to
be more inline with its actual job and return type. This commit adds
some comments into the code to make it more clear on what is occuring,
along with some class documentation comments.
